### PR TITLE
Migrate to Gradle Build and Update Workflows

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

$Subject

Related to [#5073](https://github.com/ballerina-platform/ballerina-standard-library/issues/5073)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,63 +1,16 @@
-name: Deployment
+name: Publish Release
 
 on:
-  release:
-    types: [ published ]
+  workflow_dispatch:
+  repository_dispatch:
+    types: [ stdlib-release-pipeline ]
```
Comment (NipunaRanasinghe) [inline .github/workflows/release.yml:6]: is it correct to use `stdlib-release-pipeline` dispatch event for the connectors as well?

File: .github/workflows/build-with-bal-test-native.yml [hunk 2]
```diff
@@ -2,76 +2,16 @@ name: GraalVM Check
...
-        run: bal test --graalvm --groups mongodb ./ballerina
-        env:
-          JAVA_HOME: /usr/lib/jvm/default-jvm
-          # Following credentials are only used during testing in docker container
-          MONGODB_USER: admin
-          MONGODB_PASSWORD: admin
+jobs:
+  call_stdlib_workflow:
+    name: Run StdLib Workflow
+    if: ${{ github.event_name != 'schedule' || (github.event_name == 'schedule' && github.repository_owner == 'ballerina-platform') }}
```
Comment (NipunaRanasinghe) [inline .github/workflows/build-with-bal-test-native.yml:15]: Can we omit using the word `stdlib` in connector repos, if possible?

File: .github/workflows/release.yml (hunk 1 above)
Comment (ThisaruGuruge) [reply .github/workflows/release.yml]: Missed this. Fixed with 91aca74ff7e8b9c62e5381e266cffb5b1f63e68b

File: .github/workflows/build-with-bal-test-native.yml (hunk 2 above)
Comment (ThisaruGuruge) [reply .github/workflows/build-with-bal-test-native.yml]: +1. Fixed with 91aca74ff7e8b9c62e5381e266cffb5b1f63e68b

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by MohamedSabthar — APPROVED** [review]

**Pull request merged**